src/dmd/expression.d

```diff
@@ -4827,11 +4827,19 @@ extern (C++) abstract class BinExp : Expression
         TOK thisOp = (op == TOKquestion) ? TOKcolon : op;
         if (e1.op == TOKtype || e2.op == TOKtype)
         {
-            error("incompatible types for ((%s) %s (%s)): cannot use '%s' with types", e1.toChars(), Token.toChars(thisOp), e2.toChars(), Token.toChars(op));
+            error("incompatible types for ((%s) %s (%s)): cannot use '%s' with types",
+                e1.toChars(), Token.toChars(thisOp), e2.toChars(), Token.toChars(op));
+        }
+        else if (e1.type.equals(e2.type))
+        {
+            error("invalid operation for `(%s) %s (%s)` with both operands of type `%s`",
```

Comment:
the error message is still bad. My first review comment was more accurate. Maybe:
> "the  `%s` operator is not supported for two operands of type `%s`  (`(%s) %s (%s)`)".

Reply:
I'm not sure there's much difference, but if necessary I'll update it.

Reply:
necessary